source/src/microapp/cs_MicroappProtocol.cpp

```diff
@@ -636,6 +700,65 @@ void MicroappProtocol::handleEvent(event_t & event) {
 			callback_func();
 			break;
 		}
+		case CS_TYPE::CMD_BLE_SET_MICROAPP_ISR: {
+			LOGd("MicroappProtocol: Reacting to CMD_BLE_SET_MICROAPP_ISR event");
+			TYPIFY(CMD_BLE_SET_MICROAPP_ISR) ble_cmd = *(TYPIFY(CMD_BLE_SET_MICROAPP_ISR)*)event.data; // microapp_ble_isr_init_t
+			
+			// we will register the handler in the class for first empty slot
+			for (int i = 0; i < MAX_BLE_ISR_COUNT; ++i) {
+				if (_ble_isr[i].callback == 0) {
+					LOGi("Register callback %x", ble_cmd.callback);
+					_ble_isr[i].callback = ble_cmd.callback;
+					_ble_isr[i].type = ble_cmd.type;
+					break;
+				}
+			}
+			break;
+		}
+		case CS_TYPE::EVT_DEVICE_SCANNED: {
+			if (!_microappIsScanning) 
+			{
+				LOGd("Microapp not scanning, so not forwarding scanned device");
+				break;
+			}
+			TYPIFY(EVT_DEVICE_SCANNED) dev = *(TYPIFY(EVT_DEVICE_SCANNED)*)event.data; // scanned_device_t
+			LOGd("MicroappProtocol: Reacting to EVT_DEVICE_SCANNED");
+
+			// copy scanned device info to ble_dev_t struct
+			ble_dev_t ble_dev;
+			ble_dev.addr_type = dev.addressType;
+			std::reverse_copy(dev.address, dev.address + MAC_ADDRESS_LENGTH, ble_dev.addr); // convert from little endian to big endian
+			ble_dev.rssi = dev.rssi;
+			ble_dev.dlen = dev.dataSize;
+			memcpy(ble_dev.data,dev.data,dev.dataSize);
+
+			uint16_t type = (uint16_t)CS_TYPE::EVT_DEVICE_SCANNED;
+			uintptr_t callback = 0;
+			// get callback and call
+			for (int i = 0; i < MAX_BLE_ISR_COUNT; ++i) {
+				if (_ble_isr[i].type == type) {
+					callback = _ble_isr[i].callback;
+					break;
+				}
+			}
+
+			if (callback == 0) {
+				LOGd("Callback not yet registered");
+				break;
+			}
+			LOGd("Call callback at 0x%x", callback);
+			// we have to do this through another coroutine perhaps (not the same one as loop!), for
+			// now stay on this stack
+			void (*callback_func)(ble_dev_t) = (void (*)(ble_dev_t)) callback;
```

Comment:
The callback now makes a copy of ble_dev_t.

Reply:
It would be preferable to pass a pointer to the struct to the microapp? As in
`void (*callback_func)(ble_dev_t*) = (void (*)(ble_dev_t*)) callback;`